packages/webview_flutter/webview_flutter_android/lib/src/android_webview.dart

```diff
@@ -1121,6 +1133,14 @@ class WebChromeClient extends JavaObject {
   /// mode.
   final HideCustomViewCallback? onHideCustomView;
 
+  final Future<void> Function(String message)? onHandleJavaScriptAlert;
```

Comment:
These should have the same name as the corresponding Java method. (e.g. `onJsAlert`). They should also have a return value of void since they can't return a synchronous value.

Reply:
Sorry, I cannot fully understand on your comment.

> They should also have a return value of void since they can't return a synchronous value.

They should have a Future value to pass the value for JSResult and it works well in my test.

![Simulator Screen Recording - iPhone 13 mini - 2023-09-28 at 15 04 27](https://github.com/flutter/packages/assets/160281/532d85fd-de1d-47c6-b5e6-cdbf9f6fabc0)

FileChooser in android_webview also return a List<String> previously, So I just follow it to implement.

Can you explain more to fix this?
I really appreciate your help :)

Reply:
Ah, now I understand. Your implementation is correct. I didn't notice the `JSResult` in the params.